src/date-input/stand-alone-input.tsx

```diff
@@ -166,13 +164,20 @@ export const StandAloneInput = ({
         // remove the cursor if it
         // - field is 'none'
         // - error happened from range value check
-        if (focusType === "none" || isError) {
+        if (focusType === "none") {
```

Comment:
i think this useEffect is no longer needed as the blur event is now handled by the parent. `isError` is also no longer required?